estimagic/optimization/optimize.py

```diff
@@ -1,379 +1,869 @@
-        gradient_kwargs (dict): Additional keyword arguments for the gradient.
-        gradient_options (dict): Options for the gradient function.
-        logging (str or pathlib.Path or list): Path(s) to (an) sqlite3 file(s) which
-            typically has the file extension ``.db``. If the file does not exist,
-            it will be created. See :ref:`logging` for details.
-        log_options (dict or list of dict): Keyword arguments to influence the logging.
-            See :ref:`logging` for details.
-        dashboard (bool): Whether to create and show a dashboard, default is False.
-            See :ref:`dashboard` for details.
-        dash_options (dict or list of dict, optional): Options passed to the dashboard.
-            Supported keys are:
-                - port (int): port where to display the dashboard.
-                - no_browser (bool): whether to display the dashboard in a browser.
-                - rollover (int): how many iterations to keep in the convergence plots.
-    Returns:
-        results (tuple or list of tuples): Each tuple consists of the harmonized result
-        info dictionary and the params DataFrame with the minimizing parameter values
-        of the untransformed problem as specified of the user.
-    """
-    # Set a flag for a maximization problem.
-    general_options = {} if general_options is None else general_options
-    general_options["_maximization"] = True
+        criterion (callable): A function that takes a pandas DataFrame (see
+            :ref:`params`) as first argument and returns one of the following:
 
-    if isinstance(gradient, list):
-        gradient = [negative_gradient(grad) for grad in gradient]
-    else:
-        gradient = negative_gradient(gradient)
+            - scalar floating point or a :class:`numpy.ndarray` (depending on the
+              algorithm)
+            - a dictionary that contains at the entries "value" (a scalar float),
+              "contributions" or "root_contributions" (depending on the algortihm) and
+              any number of additional entries. The additional dict entries will be
+              logged and (if supported) displayed in the dashboard. Check the
+              documentation of your algorithm to see which entries or output type are
+              required.
+        params (pandas.DataFrame): A DataFrame with a column called "value" and optional
+            additional columns. See :ref:`params` for detail.
+        algorithm (str or callable): Specifies the optimization algorithm. For supported
+            algorithms this is a string with the name of the algorithm. Otherwise it can
+            be a callable with the estimagic algorithm interface. See :ref:`algorithms`.
+        criterion_kwargs (dict): Additional keyword arguments for criterion
+        constraints (list): List with constraint dictionaries.
+            See .. _link: ../../docs/source/how_to_guides/how_to_use_constranits.ipynb
```

Comment:
This does not work.

Reply:
Fixed this and other warnings in the documentation. Thanks!